cads_adaptors/tools/url_tools.py

```diff
@@ -39,62 +80,50 @@ def requests_to_urls(
 
 
 def try_download(
-    urls: List[str], context: Context, server_suggested_filename=False, **kwargs
-) -> List[str]:
+    urls: list[str],
+    context: Context,
+    server_suggested_filename: bool = False,
+    # TODO: Check with ECMWF these parameters before merging
+    maximum_tries: int = 10,
+    retry_after: float | tuple[float, float, float] = (1, 120, 1.3),
+    # the default timeout value (3) has been determined empirically (it also included a safety margin)
+    timeout: float = 3,
```

Comment:
@EddyCMWF I'm not 100% sure I got the default settings you'd like to use right.
Also, the incremental `retry_after` needs https://github.com/ecmwf/multiurl/pull/30